packages/alfa-math/src/matrix.ts

```diff
@@ -1,63 +1,175 @@
-import * as math from "mathjs";
-import { Vector } from "./vector";
+import { Real } from "./real";
 
-export type Matrix<M extends number = number, N extends number = number> = [
-  Vector<N>,
-  ...Array<Vector<N>>
-] & { readonly length: M };
+/**
+ * @see https://en.wikipedia.org/wiki/Matrix_(mathematics)
+ */
+export type Matrix = Array<Array<number>>;
```

Comment:
I initially converted both the `Matrix` and `Vector` types to classes with constructors that made sanity checks like that. The issue was that they became extremely tricky to use as they lost the ability to use index notation for reading and assignment.

The previous tuple hacks used to kinda-sorta enforce strict lengths didn't really work out in practice as well as they required explicit annotation to avoid the lengths being inferred as just `number`. TypeScript really doesn't support these kinds of dependant types 😕